# HG changeset patch
# User Sergey Plotnikov <seryozha.plotnikov@gmail.com>
# Date 1392744115 -3600
#      Tue Feb 18 18:21:55 2014 +0100
# Branch gui-release
# Node ID dca6ecfd622cfb0ab21492cd9ee86cba65d7d76a
# Parent  fb96b7f5524295c2e538106f6fe57746a8e12aba
findobj.m: let leading negation work (bug #41588).

* findobj.m: keep track of extranegation.

diff --git a/scripts/plot/util/findobj.m b/scripts/plot/util/findobj.m
--- a/scripts/plot/util/findobj.m
+++ b/scripts/plot/util/findobj.m
@@ -114,25 +114,29 @@ function h = findobj (varargin)
     args = varargin(n1 : nargin);
   else
     args = {};
   endif
 
   regularexpression = [];
   property          = [];
   logicaloperator   = {};
+  extranegation     = [];
   pname             = {};
   pvalue            = {};
   np = 1;
   na = 1;
   operatorprecedence = {"-not", "-and", "-or", "-xor"};
 
   while (na <= numel (args))
     regularexpression(np) = 0;
     property(np) = 0;
+    if (numel (extranegation) < np)
+      extranegation(np) = false;
+    endif
     logicaloperator{np} = "and";
     if (ischar (args{na}))
       if (strcmpi (args{na}, "-regexp"))
         if (na + 2 <= numel (args))
           regularexpression(np) = 1;
           na = na + 1;
           pname{np} = args{na};
           na = na + 1;
@@ -172,20 +176,18 @@ function h = findobj (varargin)
           else
             logicaloperator{np} = "and";
           endif
           np = np + 1;
         else
           error ("findobj: inconsistent number of arguments");
         endif
       else
-        ## This is sloppy ... but works like Matlab.
         if (strcmpi (args{na}, "-not"))
-          h = [];
-          return;
+          extranegation(np) = true;
         endif
         na = na + 1;
       endif
     else
       error ("findobj: properties and options must be strings");
     endif
   endwhile
 
@@ -236,16 +238,19 @@ function h = findobj (varargin)
               endif
             else
               match(nh,np) = false;
             endif
           endif
         else
           match(nh,np) = false;
         endif
+        if (extranegation(np))
+          match(nh,np) = ! match(nh,np);
+        endif
       endfor
     endfor
 
     if (numpairs > 1)
       for no = 1 : numel (operatorprecedence)
         pairs = find (strcmp (logicaloperator(2:end), ...
                               operatorprecedence{no}(2:end)));
         for np = sort (pairs, "descend")
